strawberry/schema/execute.py

```diff
@@ -66,27 +66,27 @@ async def execute(
         # Note: In graphql-core the schema would be validated here but in
         # Strawberry we are validating it at initialisation time instead
 
-        try:
-            async with extensions_runner.parsing():
+        async with extensions_runner.parsing():
+            try:
                 if not execution_context.graphql_document:
                     execution_context.graphql_document = parse_document(query)
-        except GraphQLError as error:
-            execution_context.errors = [error]
-            return ExecutionResult(
-                data=None,
-                errors=[error],
-                extensions=await extensions_runner.get_extensions_results(),
-            )
-
-        except Exception as error:  # pragma: no cover
-            error = GraphQLError(str(error), original_error=error)
-
-            execution_context.errors = [error]
-            return ExecutionResult(
-                data=None,
-                errors=[error],
-                extensions=await extensions_runner.get_extensions_results(),
-            )
+            except GraphQLError as error:
+                execution_context.errors = [error]
+                return ExecutionResult(
+                    data=None,
+                    errors=[error],
+                    extensions=await extensions_runner.get_extensions_results(),
+                )
+
+            except Exception as error:  # pragma: no cover
+                error = GraphQLError(str(error), original_error=error)
+
+                execution_context.errors = [error]
+                return ExecutionResult(
+                    data=None,
+                    errors=[error],
+                    extensions=await extensions_runner.get_extensions_results(),
+                )
```

Comment:
Noticed while investigating #3204:

Looks like this block should not have been moved or kept at all. Now it's catching `Exception`s raised by `parse_document`, which there shouldn't be any.

Before it was moved, it used to catch errors raised inside extensions during the parsing step and add those to the execution result. Such errors are usually progamming errors which we might prefer to cause an internal server error instead.